pkg/controller/migration/convert/network.go

```diff
@@ -437,16 +440,213 @@ func getCNIPlugin(c *components) (operatorv1.CNIPluginType, error) {
+	c.node.ignoreEnv("calico-node", "CALICO_IPV6POOL_NAT_OUTGOING")
+	c.node.ignoreEnv("calico-node", "CALICO_IPV6POOL_NODE_SELECTOR")
 
 	return nil
 }
+
+// getIPPools searches through the pools passed in using the matcher function passed in to see if the pool
+// should be selected, the first pool that the matcher returns true on is returned.
+// If there is an error returned from the matcher then that error is returned.
+// If no pool is found and there is no error then nil,nil is returned.
+func getIPPool(pools []crdv1.IPPool, matcher func(crdv1.IPPool) (bool, error)) (*crdv1.IPPool, error) {
+	for _, pool := range pools {
+		if pool.Spec.Disabled {
+			continue
+		}
+		yes, err := matcher(pool)
+		if err != nil {
+			return nil, err
+		}
+		if yes {
+			return &pool, nil
+		}
+	}
+	return nil, nil
+}
+
+// isIPv4 check if the IP is an IPv4 address
+func isIpv4(ip net.IP) bool {
+	return ip.To4() != nil
+}
+
+// isIPv6 checks if the IP is an IPv6 address
+func isIpv6(ip net.IP) bool {
+	return ip.To4() == nil
+}
+
+// selectInitialPool searches through pools for enabled pools, returning the
+// first to match one of the following:
+//   1. the passed in cidr
+//   2. one prefixed with default and matching the isver IP version
+//   3. one matching isver IP version
+// if none match then nil, nil is returned
+// if there is an error parsing the cidr in a pool then that error will be returned
+func selectInitialPool(pools []crdv1.IPPool, cidr *string, isver func(ip net.IP) bool) (*crdv1.IPPool, error) {
+	// Get IP pool if there is one that matches the initial CIDR
+	if cidr != nil {
+		pool, _ := getIPPool(pools, func(p crdv1.IPPool) (bool, error) { return p.Spec.CIDR == *cidr, nil })
+		if pool != nil {
+			return pool, nil
+		}
+	}
+	// If we don't have a pool then try finding one with the right version that is prefixed with 'default'
+	pool, err := getIPPool(pools, func(p crdv1.IPPool) (bool, error) {
+		ip, _, err := net.ParseCIDR(p.Spec.CIDR)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse IPPool %s in datastore: %s", p.Name, err.Error())
+		}
+		if isver(ip) {
+			if strings.HasPrefix(p.Name, "default") {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	if pool != nil {
+		return pool, nil
+	}
+
+	// If we don't have a pool then just grab any that has the right version
+	pool, err = getIPPool(pools, func(p crdv1.IPPool) (bool, error) {
+		ip, _, err := net.ParseCIDR(p.Spec.CIDR)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse IPPool %s in datastore: %s", p.Name, err.Error())
+		}
+		return isver(ip), nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	if pool != nil {
+		return pool, nil
+	}
+	return nil, nil
+}
+
+// convertPool converts the src (CRD) pool into an Installation/Operator IPPool
+func convertPool(src crdv1.IPPool) (*operatorv1.IPPool, error) {
+	p := operatorv1.IPPool{CIDR: src.Spec.CIDR}
+
+	ip := src.Spec.IPIPMode
```

Comment:
If an empty string mode means never, i think you could calm this code down a lot by setting it to that up front:
```suggestion
	ip := src.Spec.IPIPMode
	if ip == "" {
		ip = crdv1.IPIPModeNever
	}
```